### DISCUSSION TITLE
AttributeError exception when raising HTTPException using headers parameter

### DISCUSSION THREAD

**Discussion by DorAbu** [fastapi/fastapi#8562] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, HTTPException

app = FastAPI()


@app.get("/")
async def root():
    raise HTTPException(
        status_code=503,
        detail=f"The server is not ready to handle the request, due to ...",
        headers={"Retry-After": 20}
    )
```

### Description

In HTTPException object signature (attached below 1.),  headers type hint is Optional[Dict[str, Any]].
But for example, when Dict[str, int] is given as in the "example code" section, the implementation in Starlette will raise an exception - AttributeError: 'int' object has no attribute 'encode' (attached below 2.).

Seems that Starlette implementation expects Optional[Dict[str, str]] type hint, FastAPI may resolve it by converting the values of the headers to strings.

1. ![image](https://user-images.githubusercontent.com/34128882/150327997-3b775b86-fd71-41c1-9016-6382cb44de78.png)
2. ![image](https://user-images.githubusercontent.com/34128882/150328167-ba9f298f-03ad-419d-bb63-b749d31d3d3d.png)

### Operating System

Windows

### FastAPI Version

0.70.1

### Python Version

3.8.9 (tags/v3.8.9:a743f81, Apr  6 2021, 14:02:34) [MSC v.1928 64 bit (AMD64)]

### Additional Context

starlette==0.16.0

**Comment by harunyasar** [comment]
I am a bit confused. This function signature below clearly shows that it is supposed to be a `str`.

https://github.com/encode/starlette/blob/34d9f0f10f2e4706551acf340eb1eaf4d8c87b49/starlette/responses.py#L59

**Reply by DorAbu** [reply]
Yes but I referred to the signature of the HTTPException object, please check the attached print screen

**Comment by tiangolo** [comment, marked as answer]
Indeed! This is wrong. The type annotation should define that it expects a `str` as value. I just fixed it in https://github.com/tiangolo/fastapi/pull/9648

The fix will be available in the next version, released in a few hours (the current version is `0.96.0`).
